test/weighted_matching_test.cpp

```diff
@@ -0,0 +1,173 @@
+//=======================================================================
+// Copyright (c) 2018 Yi Ji
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+//=======================================================================
+
+#include <boost/graph/max_cardinality_matching.hpp>
+#include <boost/graph/maximum_weighted_matching.hpp>
+
+#include <iostream>                      // for std::cout
+#include <boost/property_map/vector_property_map.hpp>
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/adjacency_matrix.hpp>
+#include <boost/graph/random.hpp>
+#include <ctime>
+#include <boost/random.hpp>
+#include <boost/test/minimal.hpp>
```

Comment:
I strongly recommend that you don't use randomly generated test cases here. Using them as a separate utility to discover bugs is up to you, but the test cases here should attempt to provide good coverage of the algorithm by providing inputs that execute different code paths, edge cases, etc.
Then you also don't need to run a brute-force verification: just store the answer and compare.

Reply:
Nice advice, thank you. I will remove brute-force verification and use pre-computed answers instead.